### PR TITLE
[MacToolbar] Have the search shortcut in the search bar

### PR DESCRIPTION
**Pull request by iainx**
Fixes BXC #38945`

### CODE REVIEW

File: main/src/addins/MacPlatform/MainToolbar/SearchBar.cs [hunk 1]
```diff
@@ -155,6 +155,8 @@ public override void SelectWithFrame (CGRect aRect, NSView inView, NSText editor
 			}
 		}
 
+		static readonly string firstResponderPlaceholder = "Search";
+		static readonly string placeholder = "Press '⌘.' to search";
```
Comment (Therzok) [inline main/src/addins/MacPlatform/MainToolbar/SearchBar.cs]: This is not the right fix.

We have [this](https://github.com/iainx/monodevelop/blob/fix-38945/main/src/addins/MacPlatform/MainToolbar/MainToolbar.cs#L318-L321)

Comment (Therzok) [inline main/src/addins/MacPlatform/MainToolbar/SearchBar.cs]: This comes from the keybinding service and handles setting the text to the correct one. If we want to sanitize it to show a nice ⌘ symbol, sure.

Comment (iainx) [inline main/src/addins/MacPlatform/MainToolbar/SearchBar.cs]: Oh, how in my refactoring of the toolbar did I never notice that...

**Comment by iainx** [comment]
Fixed

File: main/src/addins/MacPlatform/MainToolbar/SearchBar.cs [hunk 2]
```diff
@@ -155,6 +155,18 @@ public override void SelectWithFrame (CGRect aRect, NSView inView, NSText editor
...
 		}
 
+		static readonly string firstResponderPlaceholder = "Search";
+		string placeholderText = "Search";
+		public string PlaceholderText {
+			get {
+				return placeholderText;
+			}
+			set {
+				placeholderText = value ?? "Search";
```
Comment (slluis) [inline main/src/addins/MacPlatform/MainToolbar/SearchBar.cs]: This "Search" text should be localizable

Comment (Therzok) [inline main/src/addins/MacPlatform/MainToolbar/SearchBar.cs]: It is. `MakePlaceholderString` calls `GettextCatalog.GetString`.

Comment (slluis) [inline main/src/addins/MacPlatform/MainToolbar/SearchBar.cs]: The GetString call in MakePlaceholderString is useless if the string is not marked as translatable, and it won't be marked as such unless there is an explicit GettextCatalog.GetString("Search") call.

Comment (iainx) [inline main/src/addins/MacPlatform/MainToolbar/SearchBar.cs]: I've fixed this up

Comment (slluis) [inline main/src/addins/MacPlatform/MainToolbar/SearchBar.cs]: It's not fixed, you need to add a GettextCatalog.GetString("Search") call.

Comment (Therzok) [inline main/src/addins/MacPlatform/MainToolbar/SearchBar.cs]: https://github.com/mono/monodevelop/pull/1287/files

**Pull request merged**